kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionCoordinator.java

```diff
@@ -647,6 +663,10 @@ public void fail(Errors errors) {
                             if (epochAndMetadata.getCoordinatorEpoch() == coordinatorEpoch) {
                                     // This was attempted epoch fence that failed, so mark this state on the metadata
                                 epochAndMetadata.getTransactionMetadata().setHasFailedEpochFence(true);
+
+                                // this line is not present in Kafka code base ?
```

Comment:
This is a pretty interesting observation to me. It'd be great to understand why we diverge here. Is it because the writes on the kafka broker are to the local disk and that essentially never fails?

Reply:
Just noticed you mention the same thing in your PR description. I think we agree on the likely reason for this missing state transition.